src/renderer/components/inputs/NumberInput.tsx

```diff
@@ -27,6 +29,22 @@ export const getPrecision = (n: number) => {
     return Math.min(10, n.toFixed(15).replace(/0+$/, '').split('.')[1]?.length ?? 0);
 };
 
+const clampMax = (value: number, max_val: number | undefined | null) => {
+    if (max_val !== undefined && max_val !== null) {
+        const valOrMax = Math.min(value, max_val);
+        return valOrMax;
+    }
+    return value;
+};
+
+const clampMin = (value: number, min_val: number | undefined | null) => {
+    if (min_val !== undefined && min_val !== null) {
+        const valOrMin = Math.max(value, min_val);
+        return valOrMin;
+    }
+    return value;
+};
+
```

Comment:
Could we combine this into a single `clmap` function?

Reply:
Yep

Reply:
I'll do that separately though since I'm pretty sure I'll need to do that in the slider as well